### PR TITLE
Add configurable s3Key as an optional parameter.

### PR DESCRIPTION
**Pull request by rbugatha**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by adam-aws — APPROVED** [review]

**Review by rbugatha — COMMENTED** [review]
submitting for review

File: src/main/java/software/amazon/payloadoffloading/S3BackedPayloadStore.java [hunk 1]
```diff
@@ -20,8 +20,8 @@ public S3BackedPayloadStore(S3Dao s3Dao, String s3BucketName) {
     }
 
     @Override
-    public String storeOriginalPayload(String payload, Long payloadContentSize) {
-        String s3Key = UUID.randomUUID().toString();
+    public String storeOriginalPayload(String payload, Long payloadContentSize, String... optional) {
+        String s3Key = optional.length > 0 ? optional[0] : UUID.randomUUID().toString();;
```
Comment (eddy-aws) [inline src/main/java/software/amazon/payloadoffloading/S3BackedPayloadStore.java:24]: Extra sem-colon?

File: src/main/java/software/amazon/payloadoffloading/PayloadStore.java [hunk 2]
```diff
@@ -12,15 +12,15 @@
...
      * @return a pointer that must be used to retrieve the original payload later.
      * @throws AmazonClientException  If any internal errors are encountered on the client side while
      *                                attempting to make the request or handle the response. For example
      *                                if a network connection is not available.
      * @throws AmazonServiceException If an error response is returned by actual PayloadStore indicating
      *                                either a problem with the data in the request, or a server side issue.
      */
-    String storeOriginalPayload(String payload, Long payloadContentSize);
-
+    String storeOriginalPayload(String payload, Long payloadContentSize, String... optional);
```
Comment (eddy-aws) [inline src/main/java/software/amazon/payloadoffloading/PayloadStore.java:23]: Why don't we create a new method with a customS3Key as a third argument?

File: src/main/java/software/amazon/payloadoffloading/S3BackedPayloadStore.java [hunk 3]
```diff
@@ -21,7 +21,12 @@ public S3BackedPayloadStore(S3Dao s3Dao, String s3BucketName) {
 
     @Override
     public String storeOriginalPayload(String payload, Long payloadContentSize) {
-        String s3Key = UUID.randomUUID().toString();
+        return storeOriginalPayload(payload, payloadContentSize, null);
+    }
+
+    @Override
+    public String storeOriginalPayload(String payload, Long payloadContentSize, String key) {
+        String s3Key = (key!=null) ? key : UUID.randomUUID().toString();;
```
Comment (adam-aws) [inline src/main/java/software/amazon/payloadoffloading/S3BackedPayloadStore.java:29]: double semicolon

**Pull request merged**